File: week4/crawllinks.py
# Python 3
"""This will crawllinks"""
import urllib.request
from bs4 import BeautifulSoup

LINKPOS = 18
REPS = 7
URL = 'http://python-data.dr-chuck.net/known_by_Kinnon.html'

def crawllinks(url, linkpos, reps):
    """Will crawl"""
    html = urllib.request.urlopen(url).read()
    soup = BeautifulSoup(html, 'lxml')
    tags = soup('a')

    new_url = tags[linkpos - 1]['href']
    reps -= 1
    if reps > 0:
        return crawllinks(new_url, linkpos, reps)
    else:
        return tags[linkpos - 1].string

print(crawllinks(URL, LINKPOS, REPS))

File: week4/crawllinks.js
//  require cheerio
const cheerio = require('cheerio')
const request = require('request')

const LINKPOS = 18
const REPS = 7
const URL = 'http://python-data.dr-chuck.net/known_by_Kinnon.html'

function crawllinks(url, linkpos, reps) {
  request(url, function (error, response, body) {
    if (!error && response.statusCode == 200) {
      const $ = cheerio.load(body)

      let index = linkpos - 1
      let tag = $('a').get(index)
      let new_url = tag.attribs.href
      let text = tag.children[0].data

      reps -= 1

      // return console.log because async nature of requests will not return a 
      // proper value to function call. Logs 'undefined'
      return reps > 0 ? crawllinks(new_url, linkpos, reps) : console.log(text)
    }
  })
}

crawllinks(URL, LINKPOS, REPS)